device_specific/flash_the_ttgo_tdisplay

```diff
@@ -81,25 +93,183 @@ fi
+  [ "${CI:-false}" = false ] && echo "This should only happen within ci builds, check manually" && exit 1
+  enable_secure_boot=n
+fi
+
+[ -f sdkconfig ] && mv sdkconfig sdkconfig.bak # make sure sdkconfig is removed from previous builds
+
 cp configs/sdkconfig_display_ttgo_tdisplay.defaults sdkconfig.defaults
-sed -i.bak '/CONFIG_DEBUG_MODE/d' ./sdkconfig.defaults
+sed -i.bak '/CONFIG_DEBUG_MODE/d' sdkconfig.defaults
 sed -i.bak '1s/^/CONFIG_LOG_DEFAULT_LEVEL_NONE=y\n/' sdkconfig.defaults
+
+if [[ "$enable_secure_boot" == "y" ]]; then
+  clear
+  echo -e "\nWARNING: You are about to enable Secure Boot on your device. This process is irreversible. Once Secure Boot is enabled, all future installations, updates, or firmware modifications will require the signing key created during this process. It is crucial to store the signing key (~/jade-diy.pem) in a secure, offline location and remove it from your computer to prevent unauthorized access.\n\n"
+  echo -e "Please ensure that you have backed up the signing key and stored it securely offline. Failure to securely store this key could result in being unable to update or modify your device in the future.\n\n"
+  echo -e "The process will begin in 10 seconds. Press Ctrl+C to cancel."
+  for i in {10..1}; do
+    printf "\rStarting in %d seconds..." "$i"
+    sleep 1
+  done
+
+  if [ ! -f ~/jade-diy.pem ]; then
+    openssl genrsa -out ~/jade-diy.pem 3072
+  fi
+
+  sed -i.bak '/CONFIG_EFUSE_VIRTUAL/d' sdkconfig.defaults
+
+  sed -i.bak '/CONFIG_ESP32_DISABLE_BASIC_ROM_CONSOLE=/d' sdkconfig.defaults
+  echo "CONFIG_ESP32_DISABLE_BASIC_ROM_CONSOLE=y" >> sdkconfig.defaults
+
+  sed -i.bak '/CONFIG_SECURE_DISABLE_ROM_DL_MODE=/d' sdkconfig.defaults
+  echo "CONFIG_SECURE_DISABLE_ROM_DL_MODE=y" >> sdkconfig.defaults
+
+  sed -i.bak '/CONFIG_SECURE_BOOT_SIGNING_KEY=/d' sdkconfig.defaults
+  echo 'CONFIG_SECURE_BOOT_SIGNING_KEY="../../../jade-diy.pem"' >> sdkconfig.defaults
+
+  sed -i.bak '/CONFIG_SECURE_BOOT=/d' sdkconfig.defaults
+  echo "CONFIG_SECURE_BOOT=y" >> sdkconfig.defaults
+
+  sed -i.bak '/CONFIG_SECURE_FLASH_ENC_ENABLED=/d' sdkconfig.defaults
+  echo "CONFIG_SECURE_FLASH_ENC_ENABLED=y" >> sdkconfig.defaults
+
+  sed -i.bak '/CONFIG_SECURE_FLASH_ENCRYPTION_MODE_RELEASE=/d' sdkconfig.defaults
+  echo "CONFIG_SECURE_FLASH_ENCRYPTION_MODE_RELEASE=y" >> sdkconfig.defaults
+
+  sed -i.bak '/CONFIG_ESP32_REV_MIN_/d' sdkconfig.defaults # remove all REV_MIN entries
+  echo "CONFIG_ESP32_REV_MIN_3=y" >> sdkconfig.defaults
+
+  echo "Secure Boot configuration has been updated in sdkconfig.defaults... building bootloader"
+  idf.py bootloader
+fi
 rm sdkconfig.defaults.bak
 
+# workaround to prevent changed idf manifest hash to flag build as dirty see https://github.com/Blockstream/Jade/issues/98
+git restore dependencies.lock
+
 idf.py build
 
 [ "${CI:-false}" = true ] && echo "Exiting the script for CI runners." && exit 0
 
-while [ ! -c "${tty_device}" ]; do
```

Comment:
Why did you remove these 8 lines?

Reply:
I think i did not remove them but moved them more to the top L64-74 (but skipped them within ci runners) to make sure that all flashing requirements are met before the build process starts